aten/src/ATen/native/xnnpack/Convolution.cpp

```diff
@@ -0,0 +1,302 @@
+         (padding[Layout::Parameter::Width] >= 0) &&
+         // Stride
+         (stride[Layout::Parameter::Height] > 0) &&
+         (stride[Layout::Parameter::Width] > 0) &&
+         // Dilation
+         (dilation[Layout::Parameter::Height] > 0) &&
+         (dilation[Layout::Parameter::Width] > 0) &&
+         // Groups
+         (groups > 0) &&
+         // Transpose
+         !transposed &&
+         // Input
+         (weight.size(Layout::Filter::Input) > 0) &&
+         // Output
+         (weight.size(Layout::Filter::Output) > 0) &&
+         // Output - Groups
+         ((weight.size(Layout::Filter::Output) % groups) == 0) &&
+         // Output Min / Max
+         (output_min.isIntegral(true) || output_min.isFloatingPoint()) &&
+         (output_max.isIntegral(true) || output_max.isFloatingPoint()) &&
+         (output_max.to<float>() > output_min.to<float>());
+}
+
+Context create(
+    const Tensor& weight,
+    const c10::optional<Tensor>& bias,
+    const IntArrayRef padding_,
+    const IntArrayRef stride_,
+    const IntArrayRef dilation_,
+    const int64_t groups,
+    const bool transposed,
+    const Scalar output_min,
+    const Scalar output_max) {
+  const auto padding = expand_param_if_needed(padding_, "padding", 2);
+  const auto stride = expand_param_if_needed(stride_, "stride", 2);
+  const auto dilation = expand_param_if_needed(dilation_, "dilation", 2);
+  const Tensor weight_nhwc = weight.contiguous(MemoryFormat::ChannelsLast);
+
+  TORCH_CHECK(
+      available(
+          weight_nhwc,
+          bias,
+          padding,
+          stride,
+          dilation,
+          groups,
+          transposed,
+          output_min,
+          output_max),
+      "xnnpack::convolution not available!");
+
+  xnn_operator_t convolution_op{};
+
+  const xnn_status create_status = xnn_create_convolution2d_nhwc_f32(
+      padding[Layout::Parameter::Height],                             // input_padding_top
+      padding[Layout::Parameter::Width],                              // input_padding_right
+      padding[Layout::Parameter::Height],                             // input_padding_bottom
+      padding[Layout::Parameter::Width],                              // input_padding_left
+      weight_nhwc.size(Layout::Filter::Height),                       // kernel_height
+      weight_nhwc.size(Layout::Filter::Width),                        // kernel_width
+      stride[Layout::Parameter::Height],                              // subsampling_height
+      stride[Layout::Parameter::Width],                               // subsampling_width
+      dilation[Layout::Parameter::Height],                            // dilation_height
+      dilation[Layout::Parameter::Width],                             // dilation_width
+      groups,                                                         // groups
+      weight_nhwc.size(Layout::Filter::Input),                        // group_input_channels
```

Comment:
Shouldn't the input channels be divided by the groups as well?

Reply:
Thanks for point that out, but no.  That got me at first as well.  Apparently the dimensionality of the weight tensor is such that input channels per group is already what dimension 1 represents.